listenbrainz/db/playlist.py

```diff
@@ -410,21 +417,32 @@ def create(playlist: model_playlist.WritablePlaylist) -> model_playlist.Playlist
     fields["algorithm_metadata"] = ujson.dumps(playlist.algorithm_metadata or {})
 
     with ts.engine.connect() as connection:
-        result = connection.execute(query, fields)
-        row = dict(result.fetchone())
-        playlist.id = row['id']
-        playlist.mbid = row['mbid']
-        playlist.created = row['created']
-        playlist.creator = creator['musicbrainz_id']
-        playlist.recordings = insert_recordings(connection, playlist.id, playlist.recordings, 0)
-
-        if playlist.collaborator_ids:
-            add_playlist_collaborators(connection, playlist.id, playlist.collaborator_ids)
-            collaborator_ids = get_collaborators_for_playlists(connection, [playlist.id])
-            collaborator_ids = collaborator_ids.get(playlist.id, [])
-            playlist.collaborators = get_collaborators_names_from_ids(collaborator_ids)
-
-        return model_playlist.Playlist.parse_obj(playlist.dict())
+        with connection.begin():
```

Comment:
Adding this delete code to general playlist create method feels wrong. Can we do this elsewhere? Probably just before troi runs in the spark handler or maybe with a --delete-existing flag in troi itself?

Reply:
Moved to a separate function as discussed in IRC.